substrate/frame/transaction-storage/src/lib.rs

```diff
@@ -98,10 +110,14 @@ pub mod pallet {
 			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
 			+ GetDispatchInfo
 			+ From<frame_system::Call<Self>>;
-		/// The currency trait.
-		type Currency: ReservableCurrency<Self::AccountId>;
+		/// The fungible type for this pallet.
+		type Fungible: FnMutate<Self::AccountId>
+			+ FnMutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
+			+ FnBalanced<Self::AccountId>;
```

Comment:
I am in favour of sticking to `Currency` as the naming. I know there is no consensus on this though.
The rational behind is that this type still represents a currency even though it does not directly implement a trait called currency

Reply:
done